packages/dds/matrix/src/matrix.ts

```diff
@@ -240,18 +282,29 @@ export class SharedMatrix<T = any>
 		let c = colStart;
 
 		for (const value of values) {
-			this.setCellCore(r, c, value);
+			const applied = this.setCellCore(r, c, value);
+			// If the operation is not applied then raise conflict else notify consumers of the change.
+			if (!applied) {
+				this.emit(
+					"resolveConflict",
+					this.cells.getCell(
+						this.rows.handleCache.getHandle(r),
+						this.cols.handleCache.getHandle(c),
+					), // Current value
+					value, // Ignored value
+				);
+			} else {
+				// Avoid reentrancy by raising change notifications after the op is queued.
+				for (const consumer of this.consumers.values()) {
```

Comment:
We should be consistent - setCell() always calls consumer.cellsChanged() irrespective if changes were applied or not.
Any reason not to move this call is into setCellCore and reduce code duplication?

Reply:
Actually, there will be no need to have this if we are allowing local changes one after the other by tracking clientids. So, this code will be removed.